packages/dds/tree/src/simple-tree/api/customTree.ts

```diff
@@ -141,26 +173,45 @@ export function customFromCursor<TChild>(
 					assert(reader.getFieldLength() === 1, 0xa19 /* invalid children number */);
 					const storedKey = reader.getFieldKey();
 					let key: string;
-					if (!options.useStoredKeys) {
-						const viewSchema =
-							schema.get(type) ?? fail(0xbff /* missing schema for type in cursor */);
-						if (isObjectNodeSchema(viewSchema)) {
-							const propertyKey = viewSchema.storedKeyToPropertyKey.get(storedKey);
-							if (propertyKey === undefined) {
-								assert(
-									viewSchema.allowUnknownOptionalFields,
-									0xc00 /* found unknown field where not allowed */,
-								);
-								// Skip unknown optional fields when using property keys.
-								return;
-							}
-							key = propertyKey;
-						} else {
+
+					switch (options.keys) {
+						case KeyEncodingOptions.allStoredKeys:
+							// Since this case might be inside of an unknown optional field,
+							// it must not depend on there being a view schema.
+							// Fortunately, its possible to implement this case without one.
 							key = storedKey;
-						}
-					} else {
-						key = storedKey;
+							break;
+						case KeyEncodingOptions.usePropertyKeys:
+						case KeyEncodingOptions.knownStoredKeys:
+							{
+								// Both these cases avoid traversing into unknown optional fields,
```

Comment:
```suggestion
								// Both these cases avoid traversing into unknown optional fields (see early return below),
```

Reply:
Don't bother with this update unless you are already pushing something else to this PR, but it would have sped up my comprehension of the code a little.

Reply:
It also might be worth pulling this out into a function like 

`getKeyFromOptions(schema, storedKey, options)`

that would get rid of the `let` binding and might give a clearer place to put the information captured in these code comments.